nrf52-hal-common/src/temp.rs

```diff
@@ -0,0 +1,57 @@
+//! Temperature sensor interface.
+
+use fpa::I30F2;
+use nb;
+use crate::target::TEMP;
+
+/// Integrated temperature sensor.
+pub struct Temp(TEMP);
+
+impl Temp {
+    /// Creates a new `Temp`, taking ownership of the temperature sensor's register block.
+    pub fn new(raw: TEMP) -> Self {
+        Temp(raw)
+    }
+
+    /// Starts a new measurement and blocks until completion.
+    ///
+    /// If a measurement was already started, it will be canceled.
+    pub fn measure(&mut self) -> I30F2 {
+        self.stop_measurement();
+        self.start_measurement();
+
+        nb::block!(self.read()).unwrap()
+    }
+
+    /// Kicks off a temperature measurement.
+    ///
+    /// The measurement can be retrieved by calling `read`.
+    pub fn start_measurement(&mut self) {
+        unsafe {
+            self.0.tasks_start.write(|w| w.bits(1));
+        }
+    }
+
+    /// Cancels an in-progress temperature measurement.
+    pub fn stop_measurement(&mut self) {
+        unsafe {
+            self.0.tasks_stop.write(|w| w.bits(1));
+            self.0.events_datardy.reset();
+        }
+    }
+
+    /// Tries to read a started measurement (non-blocking).
+    ///
+    /// Before calling this, `start_measurement` must be called.
+    ///
+    /// Returns the measured temperature in °C.
+    pub fn read(&mut self) -> nb::Result<I30F2, ()> {
```

Comment:
It seems like this function can't fail (except for `WouldBlock`) and `()` is not the right type to represent this. I don't think using the never type (`!`) in this position is stable yet, but `void::Void` would work. I believe `void` is already a dependency of this crate.

```suggestion
    pub fn read(&mut self) -> nb::Result<I30F2, Void> {
```
Note: This suggestion only covers the change required in this line. `Void` would also need to be imported above..

Reply:
Done :)